src/ConnectedMode.UnitTests/Migration/XmlFileCleanerTests.cs

```diff
@@ -26,10 +27,17 @@
 
 namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Migration
 {
+    // NOTE: in the event of a test failure the actual and expected results are
+    // written to files and added to the Test Results i.e. they should be
+    // accessible at the bottom of the test results pane (and automatically
+    // captured by the Azure DevOps CI build).
+
     [TestClass]
     public class XmlFileCleanerTests
     {
         private static readonly LegacySettings AnyLegacySettings = new LegacySettings("c:\\any\\root\\folder", "csharpruleset", "csharpXML", "vbruleset", "vbXML");
+     
+        public TestContext TestContext { get; set; }
```

Comment:
MSTest will set this property when running the test. It gives us access to e.g. the test-specific results directory created by MSTest at runtime. See [the docs](https://learn.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.testtools.unittesting.testcontext?view=visualstudiosdk-2022) for more info.